### DISCUSSION TITLE
How can I avoid Renovate raising multiple conflicting PRs?

### DISCUSSION THREAD

**Discussion by allanlewis** [renovatebot/renovate#24420] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I have a monorepo and would like to limit the number of open PRs per component to 1; otherwise merging one would require the others to be rebased, costing me CI minutes.

### What you tried so far.

I've tried making a group with `matchFileNames` set to the file for one component and `prConcurrentLimit` set to 1, but this caused all updates for that component to be grouped, which isn't what I want.

### Relevant debug logs

In summary, the logs show what I described above: Renovate is (understandably) grouping updates for each component.

**Comment by allanlewis** [comment]
In my organisation, each component is owned by a team; typically a team owns 3-5 components. Ideally I'd like Renovate to work on a round-robin basis, opening one PR for a dependency of component 1, then one for component 2, etc up to `prConcurrentLimit`.

**Comment by rarkins** [comment]
Not currently possible. I recommend you set dashboard approval true for major updates and bypass the problem that way

**Reply by allanlewis** [reply]
> set dashboard approval true for major updates

I don't see how that would help...

It's a shame this isn't possible. Any control of the order in which PRs are raised would be helpful, it seems `prPriority` doesn't help here.

**Reply by rarkins** [reply]
I may be misunderstanding what you mean. Instead of giving using specific examples you're using an abstract term "component" which is not part of Renovate vernacular

**Reply by allanlewis** [reply]
I mean I have a monorepo with a number of directories, each with a dependency spec (`pyproject.toml`/`poetry.lock` in my case). I want to prevent Renovate from raising more than one update PR per directory. I hope that makes a little more sense 🙏

**Reply by rarkins** [reply, marked as answer]
Thanks, now understood. Unfortunately still the same answer - we don't have support for this type of rate limiting. Ignore my previous discussion of dashboard approval for major

**Comment by allanlewis** [comment]
I've raised #24438 as a feature request.
